### PR TITLE
incorporating aasp code into this repo

### PR DESCRIPTION
**Pull request by stevendongatmsft**

### CODE REVIEW

File: examples/aasp/README.md
Comment (hgarvison) [inline examples/aasp/README.md:15]: Can you update the format of this README to be more step-by-step with step 1, step 2, and etc

File: cmd/aasp/main.go
Comment (SethHollandsworth) [inline cmd/aasp/main.go]: should these commented out lines be deleted?

File: cmd/aasp/main.go
Comment (SethHollandsworth) [inline cmd/aasp/main.go]: should this be getting assigned to a variable?

File: docker/aasp/Dockerfile.build
Comment (SethHollandsworth) [inline docker/aasp/Dockerfile.build]: I think this should be changed to the github.com/microsoft repo

File: docker/aasp/Dockerfile.build
Comment (SethHollandsworth) [inline docker/aasp/Dockerfile.build]: same thing here

File: examples/aasp/README.md
Comment (SethHollandsworth) [inline examples/aasp/README.md]: typo "we can generate"

File: pkg/attest/attest.go
Comment (SethHollandsworth) [inline pkg/attest/attest.go]: why change these to Printf?

**Review by SethHollandsworth — CHANGES_REQUESTED** [review]
A few quick changes, looks good!

File: cmd/aasp/main.go
Comment (stevendongatmsft) [reply cmd/aasp/main.go]: This needs to be there because it belongs to a PR https://github.com/container-investigations/aaa/pull/2 
It's temporarily commented out but will be added back once we enable the federated id thing on the cluster. I have not looked into this, but I'll look at it soon.

File: docker/aasp/Dockerfile.build
Comment (stevendongatmsft) [reply docker/aasp/Dockerfile.build]: good catch

File: cmd/aasp/main.go
Comment (stevendongatmsft) [reply cmd/aasp/main.go]: We never used it, removed.

**Comment by DomAyre** [comment]
It would be great if you could create this PR from a branch in this repo, that way Github actions can push images to a registry which allows me to run examples against your change

**Comment by stevendongatmsft** [comment]
> It would be great if you could create this PR from a branch in this repo, that way Github actions can push images to a registry which allows me to run examples against your change

Could you be more specific? What do you mean by create this PR from a branch?

**Comment by DomAyre** [comment]
Sorry for the lack of clarity, I mean instead of the Pull Request coming from a branch on your fork of this repository, you create a branch on this repository itself with your changes and make a PR from that branch.

This is because GitHub actions doesn't grant access to secrets in workflows triggered from PRs from repo forks. This is to prevent malicious actors who don't have write access on the repository making a fork, changing a workflow to leak secrets then running it.

That's why the "push X image" workflows are being skipped, if we make a PR from a branch in this repository they'll run and we'll get an image that I can do further automated testing with

File: examples/aasp/README.md
Comment (DomAyre) [inline examples/aasp/README.md:1]: As someone who hasn't come across AASP before, it doesn't seem very clear to me how this is different in principal to the key release sidecar? I can see that the key is used to encrypt a secret, and that it uses kubenetes and gRPC, but these feel more like implementation details around the core idea of storing something in mHSMs which can be released with an attestation?

I'm guessing I've missed some key detail, if the key detail is in the readme feel free to point me to it, otherwise might be worth considering elevating that key detail to the top of this document to make the difference super clear

File: examples/aasp/setup-key-mhsm.sh
Comment (DomAyre) [inline examples/aasp/setup-key-mhsm.sh:58]: Why is all this code commented out?

File: docker/aasp/Dockerfile.build
Comment (DomAyre) [inline docker/aasp/Dockerfile.build:6]: Why are we checking out the repo again? Why not just use the repo in the docker context?

File: docker/aasp/Dockerfile.build
Comment (DomAyre) [inline docker/aasp/Dockerfile.build]: Relying on a separate branch existing feels risky to me?

File: docker/aasp/Dockerfile.sample
Comment (DomAyre) [inline docker/aasp/Dockerfile.sample]: Is there a reason we're not using a package manager to download these dependencies?

File: docker/aasp/Dockerfile.sample
Comment (DomAyre) [inline docker/aasp/Dockerfile.sample]: Is there any documentation on what this should look like?

File: cmd/aasp/main.go
Comment (DomAyre) [inline cmd/aasp/main.go]: Why is this code commented out?

File: cmd/aasp/main.go
Comment (DomAyre) [inline cmd/aasp/main.go]: Commented?

File: cmd/aasp/main.go
Comment (DomAyre) [inline cmd/aasp/main.go]: Could you explain why this is needed?

File: cmd/aasp/main.go
Comment (DomAyre) [inline cmd/aasp/main.go]: Is this needed if we import httpginendpoints?

File: pkg/attest/attest.go
Comment (DomAyre) [inline pkg/attest/attest.go]: Should the code for fetching attestation reports be refactored out between here and Attest() instead of copy/pasting code?

File: pkg/attest/platform_cert_fetcher.go
Comment (DomAyre) [inline pkg/attest/platform_cert_fetcher.go:34]: Will this be addressed in this PR?

File: docker/aasp/Dockerfile.build
Comment (stevendongatmsft) [reply docker/aasp/Dockerfile.build]: yeah this will be updated.

File: pkg/attest/platform_cert_fetcher.go
Comment (stevendongatmsft) [reply pkg/attest/platform_cert_fetcher.go:34]: yes, retrieving certs from THIM will be address in this PR.

File: cmd/aasp/main.go
Comment (stevendongatmsft) [reply cmd/aasp/main.go]: This is already removed. I believe you are looking at an outdated branch.

File: cmd/aasp/main.go
Comment (stevendongatmsft) [reply cmd/aasp/main.go]: Explain why we make it a const or why we check == aasp?

File: cmd/aasp/main.go
Comment (stevendongatmsft) [reply cmd/aasp/main.go]: It is re-enabled for workload identity.

File: cmd/aasp/main.go
Comment (stevendongatmsft) [reply cmd/aasp/main.go]: this is outdated

File: pkg/attest/attest.go
Comment (stevendongatmsft) [reply pkg/attest/attest.go]: This is removed because we don't need host data.

File: examples/aasp/setup-key-mhsm.sh
Comment (stevendongatmsft) [reply examples/aasp/setup-key-mhsm.sh:58]: We just want to create a simple key release policy for the simple demonstration purposes instead of making the policy too complicated.

File: docker/aasp/Dockerfile.sample
Comment (stevendongatmsft) [reply docker/aasp/Dockerfile.sample]: I'm not sure what you mean.

File: docker/aasp/Dockerfile.sample
Comment (stevendongatmsft) [reply docker/aasp/Dockerfile.sample]: Yeah, I actually explored using package manager to download these dependencies a while back. jq is easier to install but the grpc package resides in the testing repository of the alpine edge branch. The word testing didn't feel very stable so I used wget to install grpcurl. This Dockerfile is updated now and jq is installed via apk.

File: docker/aasp/Dockerfile.build
Comment (stevendongatmsft) [reply docker/aasp/Dockerfile.build:6]: what do you mean by docker context?

File: examples/aasp/README.md
Comment (stevendongatmsft) [reply examples/aasp/README.md:15]: this is done

File: examples/aasp/README.md
Comment (BryceDFisher) [reply examples/aasp/README.md:1]: The only real difference is that AASP also implements the gRPC interface expected by Kata, and the SKR sidecar doesn't.

Comment (BryceDFisher) [reply examples/aasp/README.md:1]: AASP also has some slightly different behavior around VCEK certificate retrieval because the THIM endpoint is accessible in AKS but not ACI.

File: docker/aasp/Dockerfile.build
Comment (BryceDFisher) [reply docker/aasp/Dockerfile.build:6]: @DomAyre I concur.  I'll make a new PR changing the git clone after this PR is merged.

**Review by BryceDFisher — APPROVED** [review]

**Review by SethHollandsworth — APPROVED** [review]
lgtm

**Pull request merged**
